tiledb/tests/test_pandas_dataframe.py

```diff
@@ -1322,8 +1351,49 @@ def test_read_indexing_with_pyarrow_and_numpy_arrays(
             )
             assert_array_equal(arr.df[np.array(partial_dim_data)], expected_partial_df)
 
+            # square brackets accessor
             expected_dict = OrderedDict(
-                [("dim_a", dim_data.tolist()), ("rand", attr_data.tolist())]
+                {
+                    "rand": [
+                        attr_data[1].as_py(),
+                        attr_data[3].as_py(),
+                        attr_data[4].as_py(),
+                    ],
+                    "dim_a": [
+                        dim_data[1].as_py(),
+                        dim_data[3].as_py(),
+                        dim_data[4].as_py(),
+                    ],
+                }
+            )
+
+            assert_dict_arrays_equal(
+                arr[[dim_data[1].as_py(), dim_data[3].as_py(), dim_data[4].as_py()]],
+                expected_dict,
+            )
+            assert_dict_arrays_equal(
+                arr[
+                    np.array(
+                        [
+                            dim_data[1].as_py(),
+                            dim_data[3].as_py(),
+                            dim_data[4].as_py(),
+                        ]
+                    )
+                ],
+                expected_dict,
+            )
+            assert_dict_arrays_equal(
+                arr[
+                    pyarrow.array(
+                        [
+                            dim_data[1].as_py(),
+                            dim_data[3].as_py(),
+                            dim_data[4].as_py(),
+                        ]
+                    )
+                ],
+                expected_dict,
```

Comment:
Then simplying to something like this for readability

```suggestion
                {
                    "rand": py_attr_data,
                    "dim_a": py_dim_data,
                }
            )

            assert_dict_arrays_equal(arr[py_dim_data], expected_dict)
            assert_dict_arrays_equal(arr[np.array(py_dim_data)], expected_dict)
            assert_dict_arrays_equal(arr[pyarrow.array(py_dim_data)], expected_dict)
```